nautobot/docs/development/core/local-k8s.md

```diff
@@ -0,0 +1,125 @@
+
+## Kind Installation
+
+To install Kind on macOS:
+
+```bash
+brew install kind
+```
+
+To install Kind on Linux:
+
+```bash
+curl -fsSL \
+    "https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64" \
+    -o /usr/local/bin/kind
+chmod +x /usr/local/bin/kind
+```
+
+To create a new Kind cluster:
+
+```bash
+# Create the cluster
+kind create cluster \
+    --name=nautobot \
+    --config development/kind-config.yaml
+
+# Export the kubeconfig to access the cluster
+kind get kubeconfig --internal --name=nautobot > development/kind-kube-config
+```
+
+## Connecting Nautobot Containers
+
+This chapter will help you connect running Nautobot containers to the Kind network. It is necessary to do this after starting the Nautobot containers.
+
+Verify Nautobot containers are running:
+
+```bash
+# Start Nautobot containers if they are not running
+invoke start
+
+# Verify the containers are running
+docker ps
+```
+
+You should see something like this:
+
+```text
+CONTAINER ID   IMAGE                           COMMAND                  CREATED          STATUS          PORTS                    NAMES
+...
+1f191e104357   local/nautobot-dev:local-py3.11 "watchmedo auto-rest…"   2 minutes ago    Up 37 seconds (healthy)   0.0.0.0:8081->8080/tcp, [::]:8081->8080/tcp nautobot-celery_worker-1
+d396ed398a4c   local/nautobot-dev:local-py3.11 "/docker-entrypoint.…"   2 minutes ago    Up 2 minutes (healthy)    0.0.0.0:8080->8080/tcp, :::8080->8080/tcp nautobot-nautobot-1
+...
+```
+
+Verify the Kind Docker network name is `kind`:
+
+```bash
+docker network ls
+```
+
+You should see something like this:
+
+```text
+NETWORK ID     NAME    DRIVER    SCOPE
+...
+6de64b53be75   kind    bridge    local
+...
+```
+
+Connect running Nautobot containers to the Kind network:
+
+```bash
+docker network connect kind nautobot-nautobot-1
+docker network connect kind nautobot-celery_worker-1
+```
+
+Verify it works:
+
+```bash
+# Open shell in the running Nautobot container
+invoke cli
+
+# Install kubectl
+curl -fsSL \
+    "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" \
+    -o /usr/local/bin/kubectl
+chmod +x /usr/local/bin/kubectl
+
+# Test the connection
+export KUBECONFIG=/source/development/kind-kube-config
+kubectl cluster-info
+```
+
+You should see something like this:
+
+```text
+Kubernetes control plane is running at https://nautobot-control-plane:6443
+CoreDNS is running at https://nautobot-control-plane:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+
+To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
+```
+
+## Create a Testing Job
+
+You can verify the connection by creating the testing K8s job:
+
+```bash
+./development/create_k8s_job.py
+```
+
+To verify the job is running:
+
+```bash
+kubectl get job
+```
+
+You should see something like this:
+
+```text
+NAME       STATUS    COMPLETIONS   DURATION   AGE
+demo-job   Running   0/1           6s         6s
+```
```

Comment:
I note that this only works once - on a second call, I get:

```
root@bdfb425df2fa:/source# ./development/create_k8s_job.py 
Exception when creating job: (409)
Reason: Conflict
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'fa92c7ea-9a47-4129-b491-005f62e2b7ff', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': '224d51af-4522-4ec2-9641-4ab96d9dcecb', 'X-Kubernetes-Pf-Prioritylevel-Uid': 'f7d86f71-33eb-46ee-a80b-b2e3c5090792', 'Date': 'Fri, 27 Sep 2024 14:14:48 GMT', 'Content-Length': '216'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"jobs.batch \"demo-job\" already exists","reason":"AlreadyExists","details":{"name":"demo-job","group":"batch","kind":"jobs"},"code":409}
```

Reply:
Added a warning to delete the job.